### ISSUE TITLE
Compilation error using clang 6.0

### ISSUE THREAD

**Issue by BerndDoser** [HITS-AIN/PINK#13]
Beside many compiler warnings the error is:

```
/work/conan-clang-6/.conan/data/pybind11/2.2.2/conan/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/pybind11/cast.h:896:73: error: use 'template' keyword to treat 'type_caster' as a dependent template name
operator std::reference_wrapper< type> () { return ((subcaster).detail::type_caster< intrinsic_t< type> > ::operator subcaster_cast_op_type &()); }
                                                                        ^
                                                                        template
```

It looks like an inconsistency between clang and pybind11.

**Comment by BerndDoser** [comment]
See also https://devtalk.nvidia.com/default/topic/1048129/cuda-setup-and-installation/cannot-compile-pybind11-with-nvcc-and-clang/

**Comment by BerndDoser** [comment]
Since the issue is caused by the combination pybind11, clang, and cuda, it will be automatically fixed by the new dynamic python interface #40, because the pybind11 code must not compiled with nvcc anymore.
